### DISCUSSION TITLE
Library Improvement

### DISCUSSION THREAD

**Discussion by Acclorite** [Acclorite/book-story#88] [Ideas]
Thread for [Library Improvement milestone](https://github.com/Acclorite/book-story/milestone/2).
Feel free to share your ideas here.

**Comment by Eskuero** [comment]
I think it would be great to have the ability to add folders that would periodically be scaned for new content and added to the library.

Basically my usecase is I run a Komga server. When I won't have network I download the epub file. Then I have to manually add it to the library on your app. This works for me for other less tech savvy members of the family I would love to just add the Downloads folder like I mentioned in the first paragraph so anything they download is inmediately added.

**Reply by Acclorite** [reply]
> I think it would be great to have the ability to add folders that would periodically be scaned for new content and added to the library.

The current adding system will be reworked, as of what it will be, I don't really know. Whether I will be able to implement that or not depends on what rework will change. I will return to this question after reworking current approach.

**What I can say now is the current system is not suited for this.** It is quite resource hungry process and is sensitive to interruption.

>  anything they download is inmediately added

Immediately not possible as the file needs to be parsed before adding, at best a couple of seconds, for large books even a couple of minutes..

**Reply by Eskuero** [reply]
> Immediately not possible as the file needs to be parsed before adding, at best a couple of seconds, for large books even a couple of minutes..

Of course. Probably a poor choice of word I did, more like automatically scanned from time to time.

Thanks for the quick reply. The app looks really nice and looking forward to see what it becomes of it :)
